app/controllers/application_controller/policy_support.rb

```diff
@@ -130,24 +130,12 @@ def profile_build
   end
 
   private ############################
-
   # Assign policies to selected records of db
   def assign_policies(db)
     assert_privileges(params[:pressed])
     session[:pol_db] = db # Remember the DB
-    ids = find_checked_ids_with_rbac(db)
-    if ids.blank?
-      ids = [find_id_with_rbac(db, params[:id])]
-    end
-    if ids.empty?
-      add_flash(_("One or more %{model} must be selected to Policy assignment") % {
-        :model => Dictionary.gettext(db.to_s, :type => :model, :notfound => :titleize, :plural => true)}, :error)
-      @refresh_div = "flash_msg_div"
-      @refresh_partial = "layouts/flash_msg"
-      return
-    else
-      session[:pol_items] = ids # Set the array of tag items
-    end
+    ids = find_records_with_rbac(db, find_checked_items)
```

Comment:
@KrecekDaniel The methods you're replacing were returning ids of the records, that were saved into the `session`, but the method `find_records_with_rbac` does not return ids. 

You're trying to store the whole instances to the `session` here, please fix.

Also, the `ids` variable is not used anymore, except the assignment to `session[:pol_items], you can remove it.